src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowConnectorConfig.java

```diff
@@ -136,22 +155,62 @@ public void validateConnection(FailureCollector collector) {
     }
   }
 
-  public Object convertToValue(String fieldName, Schema fieldSchema, Map<String, Object> record) {
-    Schema.Type fieldType = fieldSchema.getType();
+  public void convertToValue(String fieldName, Schema fieldSchema, Map<String, Object> record,
+                               StructuredRecord.Builder recordBuilder) {
     Object fieldValue = record.get(fieldName);
+    if (fieldValue == null || fieldValue.toString().isEmpty()) {
+      // Set 'null' value as it is
+      recordBuilder.set(fieldName, null);
+      return;
+    }
+
+    Schema.LogicalType fieldLogicalType = fieldSchema.getLogicalType();
+    // Get values of logical types properly
+    if (fieldLogicalType != null) {
+      switch (fieldLogicalType) {
+        case DATETIME:
+          DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern(DATE_TIME_PATTERN);
+          try {
+            recordBuilder.setDateTime(fieldName, LocalDateTime.parse(fieldValue.toString(), dateTimeFormatter));
+          } catch (DateTimeParseException exception) {
+            throw new UnexpectedFormatException(
+              String.format("Field '%s' of type '%s' with value '%s' is not in ISO-8601 format.",
+                            fieldName, fieldSchema.getDisplayName(), fieldValue), exception);
+          }
+          return;
+        case DATE:
+          DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern(DATE_PATTERN);
+          recordBuilder.setDate(fieldName, LocalDate.parse(fieldValue.toString(), dateFormatter));
+          return;
+        case TIME_MILLIS:
+          DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern(TIME_PATTERN);
+          recordBuilder.setTime(fieldName, LocalTime.parse(fieldValue.toString(), timeFormatter));
+          return;
+        default:
+          throw new UnexpectedFormatException(String.format("Field '%s' is of unsupported type '%s'", fieldName,
+                                                            fieldLogicalType.name().toLowerCase()));
+      }
+    }
+
+    Schema.Type fieldType = fieldSchema.getType();
 
     switch (fieldType) {
       case STRING:
-        return convertToStringValue(fieldValue);
+        recordBuilder.set(fieldName, convertToStringValue(fieldValue));
+        return;
       case DOUBLE:
-        return convertToDoubleValue(fieldValue);
+        recordBuilder.set(fieldName, convertToDoubleValue(fieldValue));
+        return;
       case INT:
-        return convertToIntegerValue(fieldValue);
+        recordBuilder.set(fieldName, convertToIntegerValue(fieldValue));
+        return;
       case BOOLEAN:
-        return convertToBooleanValue(fieldValue);
+        recordBuilder.set(fieldName, convertToBooleanValue(fieldValue));
+        return;
       case UNION:
```

Comment:
I don't see unions in the type mapping table from the documentation. If this is only to handle nulls, nulls are already handled at the start of this method so this case should be removed.

Reply:
The UNION case is required for fields having nullable schema types. At the start of this method, the `null` check is for the field value, not the schema.

Reply:
removed the case statement and added the schema nullable check at the beginning. f58c78717ea204f30540f24a0352ce21ef7ed47b